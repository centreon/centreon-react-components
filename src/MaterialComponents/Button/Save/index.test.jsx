import React from 'react';

import { render } from '@testing-library/react';

import SaveButton from '.';

describe(SaveButton, () => {
  it('renders with a floppy icon', () => {
    const { container } = render(<SaveButton />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders with a loading indicator when loading flag is set', () => {
    const { container } = render(<SaveButton loading />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders with a check icon when the suceeded flag is set', () => {
    const { container } = render(<SaveButton succeeded />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
