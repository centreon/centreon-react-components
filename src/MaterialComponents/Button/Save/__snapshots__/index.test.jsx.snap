// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveButton renders with a check icon when the suceeded flag is set 1`] = `
<button
  aria-label="Save"
  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
  style="width: 40px; height: 40px;"
  tabindex="0"
  type="button"
>
  <span
    class="MuiButton-label"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
      />
    </svg>
  </span>
  <span
    class="MuiTouchRipple-root"
  />
</button>
`;

exports[`SaveButton renders with a floppy icon 1`] = `
<button
  aria-label="Save"
  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
  style="width: 40px; height: 40px;"
  tabindex="0"
  type="button"
>
  <span
    class="MuiButton-label"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M17 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V7l-4-4zm-5 16c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3zm3-10H5V5h10v4z"
      />
    </svg>
  </span>
  <span
    class="MuiTouchRipple-root"
  />
</button>
`;

exports[`SaveButton renders with a loading indicator when loading flag is set 1`] = `
<button
  aria-label="Save"
  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
  style="width: 40px; height: 40px;"
  tabindex="0"
  type="button"
>
  <span
    class="MuiButton-label"
  >
    <div
      class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 30px; height: 30px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </span>
  <span
    class="MuiTouchRipple-root"
  />
</button>
`;
